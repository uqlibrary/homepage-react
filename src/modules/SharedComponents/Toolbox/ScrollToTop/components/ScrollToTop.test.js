import { ScrollToTop } from './ScrollToTop';
jest.mock('react-router-dom');

function setup(testProps, isShallow = true) {
    const props = {
        children: 'Children',
        location: {},
        ...testProps,
    };

    return getElement(ScrollToTop, props, isShallow);
}
describe('ScrollToTop component', () => {
    it('should render given children', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    // it('should scrollTo to if location is changed', () => {
    //     const testLocation = {}; // {} !== {}
    //     const scrollToFn = jest.fn();
    //     global.window.scrollTo = scrollToFn;
    //     const wrapper = setup({});
    //
    //     // test if changed
    //     wrapper.setProps({
    //         location: testLocation,
    //     });
    //     expect(scrollToFn).toHaveBeenCalledWith(0, 0);
    //
    //     // test else
    //     scrollToFn.mockClear();
    //     wrapper.setProps({
    //         location: testLocation,
    //     });
    //     expect(scrollToFn).not.toBeCalled();
    // });
});
