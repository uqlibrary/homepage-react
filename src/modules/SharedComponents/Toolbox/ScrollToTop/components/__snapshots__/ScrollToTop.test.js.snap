// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop component should render given children 1`] = `"Children"`;
